test-lib-remote-openshift.sh

```diff
@@ -0,0 +1,105 @@
+# shellcheck shell=bash
+# some functions are used from test-lib.sh, that is usually in the same dir
+# shellcheck source=/dev/null
+source "$(dirname "${BASH_SOURCE[0]}")"/test-lib.sh
+
+# Set of functions for testing docker images in OpenShift using 'oc' command
+
+# A variable containing the overall test result; must be changed to 0 in the end
+# of the testing script:
+#   OS_TESTSUITE_RESULT=0
+# And the following trap must be set, in the beginning of the test script:
+#   trap ct_os_cleanup EXIT SIGINT
+OS_TESTSUITE_RESULT=1
+OS_CLUSTER_STARTED_BY_TEST=0
+
+# ct_os_set_path_oc_4 OC_VERSION
+# --------------------
+# This is a trick that helps using correct version 4 of the `oc`:
+# The input is version of the openshift in format 4.4 etc.
+# If the currently available version of oc is not of this version,
+# it first takes a look into /usr/local/oc-<ver>/bin directory,
+
+# Arguments: oc_version - X.Y part of the version of OSE (e.g. 3.9)
+function ct_os_set_path_oc_4() {
+    echo "Setting OCP4 client"
+    local oc_version=$1
+    local installed_oc_path="/usr/local/oc-v${oc_version}/bin"
+    echo "PATH ${installed_oc_path}"
+    if [ -x "${installed_oc_path}/oc" ] ; then
+        oc_path="${installed_oc_path}"
+        echo "Binary oc found in ${installed_oc_path}" >&2
+    else
+       echo "OCP4 not found"
+       return 1
+    fi
+    export PATH="${oc_path}:${PATH}"
+    oc version
+    if ! oc version | grep -q "Client Version: ${oc_version}." ; then
+        echo "ERROR: something went wrong, oc located at ${oc_path}, but oc of version ${oc_version} not found in PATH ($PATH)" >&1
+        return 1
+    else
+        echo "PATH set correctly, binary oc found in version ${oc_version}: $(command -v oc)"
+    fi
+}
+
+# ct_os_prepare_ocp4
+# ------------------
+# Prepares environment for testing images in OpenShift 4 environment
+#
+#
+function ct_os_set_ocp4() {
+  ct_os_set_path_oc_4 "4.4"
```

Comment:
This is a great catch. I will defined in `test-lib-remote-openshift.sh`. By the way. This version mean `oc` client version. Not server one. The server `hostname` is defined by `kubeconfig`.

Reply:
@hhorak Mayby I don't follow your proposal. But to be sure. Do we want to support more `oc` client versions?

Reply:
My point was only to not hard-code the value here, since changing it would mean to update the submodule for each image. Instead, when using an env. variable, we can change the version and override it e.g. in Jenkins.